network/framework/src/protocols/rpc_error.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::PeerId;
+use futures::channel::{mpsc, oneshot};
+use std::io;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum RpcError {
```

Comment:
Nice. To make this easier to land, you could add these types of refactors to a separate PR 😄 Will be easier to review.